src/components/TokenEnterAmount.tsx

```diff
@@ -11,28 +11,258 @@ import {
 import { NETWORK_NAMES } from 'src/shared/conts'
 import Colors from 'src/styles/colors'
 import { typeScale } from 'src/styles/fonts'
 import { Spacing } from 'src/styles/styles'
+import { useTokenInfo } from 'src/tokens/hooks'
 import { type TokenBalance } from 'src/tokens/slice'
+import { convertLocalToTokenAmount, convertTokenToLocalAmount } from 'src/tokens/utils'
+import { parseInputAmount } from 'src/utils/parsing'
 
 export const APPROX_SYMBOL = '≈'
 
 const BORDER_RADIUS = 12
 
-function groupNumber(value: string) {
-  return value.replace(/\B(?=(\d{3})+(?!\d))(?<!\.\d*)/g, 'group')
+/**
+ * This function formats numbers in a "1234.5678" format into the correct format according to the
+ * return value of the `getNumberFormatSettings` function.
+ * E.g.
+ *   - With decimal "." and grouping ",": 1234.5678 -> 1,234.5678
+ *   - With decimal "," and grouping ".": 1234.5678 -> 1.234,5678
+ */
+export function formatNumber(value: string) {
+  const { decimalSeparator, groupingSeparator } = getNumberFormatSettings()
+  return value
+    .replace(/\B(?=(\d{3})+(?!\d))(?<!\.\d*)/g, '_')
+    .replaceAll('.', decimalSeparator)
+    .replaceAll('_', groupingSeparator)
+}
+
+/**
+ * This function returns complete formatted value for a token (crypto) amount when it is being a
+ * converted value (in "ExchangeAmount" element).
+ */
+export function getReadableTokenAmount(bignum: BigNumber | null, token: TokenBalance) {
+  const { decimalSeparator } = getNumberFormatSettings()
+  if (bignum === null || bignum.isZero()) {
+    return ''
+  }
+
+  if (bignum.isLessThan(0.000001)) {
+    return `<0${decimalSeparator}000001 ${token.symbol}`
+  }
+
+  const grouped = formatNumber(bignum.decimalPlaces(6).toString())
+  return `${grouped} ${token.symbol}`
+}
+
+/**
+ * This function returns complete formatted value for a local (fiat) amount when it is being a
+ * converted value (in "ExchangeAmount" element).
+ */
+export function getReadableLocalAmount(
+  bignum: BigNumber | null,
+  localCurrencySymbol: LocalCurrencySymbol
+) {
+  const { decimalSeparator } = getNumberFormatSettings()
+  if (bignum === null || bignum.isZero()) {
+    return ''
+  }
+
+  if (bignum.isLessThan(0.000001)) {
+    return `<${localCurrencySymbol}0${decimalSeparator}000001`
+  }
+
+  const rounded = bignum.isLessThan(0.01) ? bignum.toPrecision(1) : bignum.toFixed(2)
+  const grouped = formatNumber(rounded.toString())
+  return `${localCurrencySymbol}${grouped}`
+}
+
+/**
+ * This hook is only used in tandem with `TokenEnterAmount` component. It provides all the necessary
+ * variables and handlers manage "enter amount" functionality, including rate calculations.
+ */
+export function useEnterAmount(props: {
+  token: TokenBalance
+  onSelectToken?: (token: TokenBalance) => void
+}) {
+  const { decimalSeparator, groupingSeparator } = getNumberFormatSettings()
+  const inputRef = useRef<RNTextInput>(null)
+  const bottomSheetRef = useRef<BottomSheetModalRefType>(null)
```

Comment:
what is the benefit of having these refs declared in this hook? this seems like an unexpected responsibility for this hook. does it make more sense for these refs to be created in the same component that creates the components, and have these refs passed in as a prop?

Reply:
@kathaypacific the idea is to provide an "out-of-the-box" hook that provides you with everything needed, including refs for the components. Yes, it doesn't reduce boilerplate as much but it reduces amount of actions you need to do or think about whenever you need to use this hook.

Reply:
ah, the `"out-of-the-box" hook that provides you with everything needed` part is exactly my concern - i think these kinds of components that try to do everything are generally difficult to maintain because as the number of places that they are used grows, the number of new things that they need to do also grows (as it is likely that each place requires something custom). it also makes naming the hook hard, which is one litmus test i use to measure how much sense an abstraction makes.

i guess i am unsure about what the benefits are of this hook knowing about the bottom sheet and inputs - naively, it seems like a component like the swap screen will need to render this hook, get the refs and callbacks from the hook, and pass it to the bottom sheet / inputs but it is the same work for the component to define the refs and callbacks itself without involving this hook. could you please elaborate on which scenarios benefit from using this hook as a middleman?

Reply:
@kathaypacific I see your point and in the context of specifically refs I definitely agree that it is more straightforward. I'll adjust this and the follow-up PRs to also have refs defined outside of the scope of the hook!